hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/Action.java

```diff
@@ -141,6 +141,16 @@ default boolean isCancelingOrCanceled() {
      */
     Optional<Integer> getLastActionStatusCode();
 
+    /**
+     * Timestamp of the action when it actually happened (e.g. could be on offline action)
```

Comment:
I've thought about the same- that way however, we need to add this timestamp in ActionStatuses in OfflineAction scenario - because currently it is added for the whole Action. In offline Action scenario all ActionStatuses are passed in single message. 
the only other thing i came up is action history - easier to sort by action#getTimestamp, other then getting all action statuses, finding terminal status and its timestamp

Reply:
All action status are placed into DB and could have timestamp?
If so required to sort actions - they could be sorted by statuses .. maybe last modified could be set (overriding automation) to the timestamp?